File: ICorteApi/Application/Interfaces/IBarberScheduleService.cs
namespace ICorteApi.Application.Interfaces;

public interface IBarberScheduleService : IService
{
    Task<TimeOnly[]> GetAvailableSlotsAsync(int barberShopId, DateOnly date, int[] serviceIds);
    Task<TopBarberShop[]> GetTopBarbersWithAvailabilityAsync(DateOnly dateOfWeek, int? take);
    Task<DateOnly[]> GetAvailableDatesForBarberAsync(int barberShopId, DateOnly dateOfWeek);
}

File: client-app/src/data/services/interfaces/IBarberScheduleService.ts
import { TopBarberShop } from "@/types/barberShop";
import { DateOnly, TimeOnly } from "@/types/date";
import { AxiosResponse } from "axios";

export interface IBarberScheduleService {
  getAvailableDatesForBarber(barberShopId: number, dateOfWeek: DateOnly): Promise<AxiosResponse<DateOnly[]>>;
  getAvailableSlots(barberShopId: number, date: DateOnly): Promise<AxiosResponse<TimeOnly[]>>;
  getTopBarbersWithAvailability(dateOfWeek: DateOnly): Promise<AxiosResponse<TopBarberShop[]>>;
}
